source/common/formatter/substitution_formatter.cc

```diff
@@ -63,8 +63,20 @@ const std::string SubstitutionFormatUtils::DEFAULT_FORMAT =
     "\"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" "
     "\"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"\n";
 
-FormatterPtr SubstitutionFormatUtils::defaultSubstitutionFormatter() {
-  return FormatterPtr{new FormatterImpl(DEFAULT_FORMAT, false)};
+const std::string SubstitutionFormatUtils::UDP_DEFAULT_FORMAT =
+    "[%START_TIME%] %UDP_CLUSTER_NAME% %UDP_DOWNSTREAM_BYTES_SENT% %UDP_DOWNSTREAM_BYTES_RECEIVED% "
+    "%UDP_DOWNSTREAM_ERRORS_SENT% %UDP_DOWNSTREAM_ERRORS_RECEIVED% "
+    "%UDP_DOWNSTREAM_DATAGRAMS_SENT% %UDP_DOWNSTREAM_DATAGRAMS_RECEIVED% "
+    "%UDP_DOWNSTREAM_SESS_TOTAL% %UDP_DOWNSTREAM_IDLE_TIMEOUT% "
+    "%UDP_UPSTREAM_NONE_HEALTHY% %UDP_DOWNSTREAM_SESS_NO_ROUTE%\n";
```

Comment:
IMO we can avoid this change in default format and I think remove the bool API variable. I would just clearly document the behavior for UDP and potentially document a suggested format string for UDP use cases.